kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/AtomicCryptorUsageCounter.java

```diff
@@ -192,4 +192,10 @@ private static long commenceDestroyForBoth(long combined) {
         return update(combined, AtomicCryptorUsageCounter::destroyCounterIfNecessary, AtomicCryptorUsageCounter::destroyCounterIfNecessary);
     }
 
+    /**
+     * @return true if has been destroyed for encrypt via {@linkplain #destroyForEncrypt()} or {@linkplain #destroyForBoth()} ()}
+     */
+    public boolean isDestroyedForEncrypt() {
+        return encryptorCount(state.get()) < 0;
```

Comment:
`0` isn't valid. The starting count is `1` and it will barf if you try to decrement to `0`.

```
// if the counter has been destroyed, it is negated, so we need to move towards the END state of -1
    private static int decrementCounter(int counter) {
        if (counter == 1 || counter == -1) {
            throw new IllegalStateException("cannot decrement at START or END, we have had more decrements than increments");
        }
        return counter > 0 ? counter - 1 : counter + 1;
    }
```

Reply:
So wont this give the wrong answer for `encryptorCount(state.get()) == 0`

Reply:
well, it's not destroyed in that case, it's in an illegal state. we could put in a throw for sanity.

Reply:
will do, also changing it to `encryptorCount(state.get()) <= END` is a bit more meaningful.